tests/unit/test_processpool.py

```diff
@@ -80,11 +83,169 @@ def test_notify_jobs_for_multiple_transfers(self):
         self.assertEqual(monitor.notify_job_complete(other_transfer_id), 1)
 
 
+class TestSubmitter(StubbedClientTest):
```

Comment:
What about a test here for the behavior where it has a job its going to split into 3 separate jobs. The first of which gets submitted, the second of which causes an error. Should the third one get submitted or skipped? 

Or maybe more simply 2 jobs. First one is an error, second one is a valid job.

Reply:
That seems reasonable. I can add it.

Reply:
Actually I am not sure if we would want a test for this. If the planner fails in submitting a specific job, it is problematic because it will be out of sync with the number of jobs the transfer monitor has been notified of. As a result, the get object workers will never decrement their count to zero and remove the temporary file because of the exception. In the end, there needs to be the guarantee that when submitting a get object job, the `Queue.put()` must never fail. Thoughts?

Reply:
Is it possible to flip the order of notifications? 
Right now we notify we'll submit N jobs before we know if those N submissions will succeed.
Would it make more sense to attempt to submit N jobs, and then notify that M jobs were added (where M is how many were actually successfully added).

Reply:
Ideally, it should be impossible for these to be out of sync.